Week-3/2_CheckAccountType/CheckAccountType.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+// import "@openzeppelin/contracts/utils/Address.sol";
+
+// contract CheckAccount {
+//     using Address for address;
+
+//     function checkAccountType (address _addr) external view returns(string memory) {
+//         if(_addr.isContract()){
+//             return "Contract Account";
+//         }
+//         return "EOA";
+//     }
+// }
+
+contract CheckAccount {
+
+    function checkAccountType (address _addr) external view returns(string memory) {
```

Comment:
Note: This is one of the primary ways to check for Contract Account and EOA. However, this check can be bypassed if the checkAccountType function is called inside the constructor of another contract.